client/extensions/woocommerce/woocommerce-services/views/carrier-accounts/dynamic-settings-form.js

```diff
@@ -0,0 +1,199 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import CompactCard from 'components/card/compact';
+import Checkbox from 'woocommerce/woocommerce-services/components/checkbox';
+import TextField from 'woocommerce/woocommerce-services/components/text-field';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import * as api from 'woocommerce/woocommerce-services/api';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+export const CheckboxFormFieldSet = ( props ) => {
+    return (
+        <>
+            <Checkbox
+                id={ props.fieldKey }
+                onChange={ props.onChange }
+                checked={ props.checked || false }
+            />
+            <label htmlFor={ props.fieldKey }><span>{ props.labelName }</span></label>
+        </>
+    );
+};
+
+export const DynamicCarrierAccountSettingsForm = ( props ) => {
+	const {
+        translate,
+    } = props;
+
+    const listOfFormFieldSet = [];
+
+    const [formFields, setFormFields] = useState({});
+    const [isSaving, setIsSaving] = useState(false);
+
+    const updateValue = ( fieldKey ) => ( newValue ) => {
```

Comment:
I'd recommend memoizing most (if not all) of these handlers. Not memoizing them has some implications on React's ability to perform DOM updates. When memoized, React can compare the previous and next values and decide whether or not the component tree needs to update.
When it's not memoized, React just decides to update the whole tree (because the handler's ref changes at each re-render).